bundler/lib/bundler/compact_index_client/updater.rb

```diff
@@ -99,8 +100,27 @@ def checksum_for_file(path)
         end
       end
 
+      def sha256_digest_for_file(path)
+        return nil unless path.file?
+        SharedHelpers.filesystem_access(path, :read) do
+          Base64.strict_encode64(SharedHelpers.digest(:SHA256).digest(File.read(path)))
+        end
+      end
```

Comment:
This should replace `checksum_for_file` above. Probably merge conflicts over time have caused this to drift.

There's[ one other usage of checksum_for_file](https://github.com/rubygems/rubygems/blob/54a7a7abc6bbb5b9c0528ad4144467be865506a1/bundler/lib/bundler/compact_index_client.rb#L98) but I think it's doing something that should be fixed with this PR, comparing a checksum with an etag.